python/paddle/hapi/model.py

```diff
@@ -598,6 +620,10 @@ def _compile_and_initialize(self, prog, mode):
                 startup_prog = self._startup_prog._prune(uninitialized)
                 self._executor.run(startup_prog)
 
+        if self._amp_level == "O2" and mode == 'train' and core.is_compiled_with_cuda(
+        ) and self._nranks <= 1:
```

Comment:
If `self._nranks > 1`, `self.model._optimizer.amp_init(place)` is also required under "O2".

Reply:
Thanks. Done. I upgraded the usage of Fleet API and fix this.
Before this, old Fleet APIs we used don't have `amp_init` function and I solved this in a wrong way.
Thanks:)